### ISSUE TITLE
Promised flag for projects to differentiate type of unallocated

### ISSUE THREAD

**Issue by rowleya** [UoMResearchIT/CapX#133]
It could be useful to break out the Unallocated category into Promised vs Not Promised demand.  This would then let things move between the categories as they are managed e.g. a researcher asks for short-term demand, we can say we can't promise it then it goes into Not Promised.  This would then let us see what our promised-but-unallocated demand is at future points and so determine how far out we should be promising new demand will be fulfilled.

**Comment by rowleya** [comment]
(I am aware that Unallocated is treated as a person, so we could just add two people I guess)

**Comment by aharwood2** [comment]
This is a good suggestion. We might be able to capture this with the provisional flag. i.e. unallocated + provisional = promised unallocated + not provisional = unallocated not promised. I'm going to create an issue to ensure that all projects which are pre-funded in the pipeline are shown as provisional but this would ruin this approach. I think I'm going to change the title of this issue to encompass that issue as well so they are thought about in the round.

**Comment by aharwood2** [comment]
I think the simples thing actually might just be to add a "promised" flag to the project itself?

**Comment by aharwood2** [comment]
I'm still not sure about the merit of this feature. If we assign someone as provisional is that not the same as promising to resource it but not necessarily commiting to resource it? Conversely if we don't assign anyone we are essentially saying we are promising to resource it but we'll try so make backup plans? Just didn't want to add another field that perhaps duplicates information. Happy to be convinced otherwise though.

**Comment by rowleya** [comment]
The aim of this was to see how far into the future we can guarantee allocations given our current resource and the sum total of unallocated tasks which might need future resource.  This assumes that there are in fact future unallocated tasks that need resource of course, which might require a bit more discussion to confirm.

Note this ticket was written against an older version of CapX which had an unallocated person.  In the current CapX this is now hidden as there isn't an "unallocated" person any more in the list.  It might be useful to have a sum total count of unallocated resources somewhere; if this is added, it would then be useful to know if this is promised resource (needs allocation but currently no-one available) or unpromised (wants allocation if someone becomes free, but otherwise project can be rejected).

**Comment by aharwood2** [comment]
Might be facilitated by #98?

**Comment by rowleya** [comment]
Yes sounds like it from the description.  Happy to wait and see!

**Comment by aharwood2** [comment]
@rowleya I'm circling back round the issues as we prep for the next release. Do you think we still need to look into this?

**Comment by rowleya** [comment]
I suspect this is covered now; I think the current statistics etc. seem to manage reasonably well without this!
